sdk/cosmosdb/azure_data_cosmos/src/clients/mod.rs

```diff
@@ -0,0 +1,5 @@
+mod cosmos_client;
+mod database_client;
+
+pub use cosmos_client::{CosmosClient, CosmosClientMethods};
+pub use database_client::{DatabaseClient, DatabaseClientMethods};
```

Comment:
Didn't consider this. I don't think we should export sub-clients from the crate root since they are not creatable directly and only "pollute" the namespace. @RickWinter @JeffreyRichter thoughts?

Reply:
Yes, I would agree. The crate root is the main stuff that all/most customers need and sub crates are for the minority of customers or are gettable from something in the root create so as not to pollute the root crate (as Heath says).

Reply:
I think that makes sense. My main concern was that `DatabaseClient` (and later `ContainerClient`) are types that users will likely want to name in their apps. For example, when a user is building a helper to fetch an item that takes `ContainerClient` as an argument. However, I think I'm sufficiently convinced that will be rare, and the type will still be exported publicly, so it can be used, it just requires referencing the submodule via `use` or a full path.

Reply:
The approach I'm going with here is to continue to export `DatabaseClient(Methods)` from _this_ module. But at `lib.rs` I'm exporting only `CosmosClient` and `CosmosClientMethods` from the top-level. That way the database client types are still nameable with `azure_data_cosmos::clients::DatabaseClient` but only `CosmosClient(Methods)` is in the top-level namespace.

Reply:
> it just requires referencing the submodule via `use` or a full path

Fortunately, rust-analyzer makes simple work of that. Too simple, in fact. In some cases I wish it wouldn't suggest some things because I think it leads to confusion but can't have my cake and eat it too.